include/sound/soc-acpi.h

```diff
@@ -200,6 +203,7 @@ struct snd_soc_acpi_mach {
 	const char *tplg_filename;
 	const char *board;
 	struct snd_soc_acpi_mach * (*machine_quirk)(void *arg);
+	bool (*machine_check)(void *arg);
```

Comment:
probably it is going to be too much overload, but the functionality of this new callback is `is_machine_match()`